src/types.ts

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+export interface Block {
+  txId: string;
+  height: number;
+}
+
+export interface DecodedScript {
+  type: string;
+  address: string;
+  timelock?: number;
+  value?: number;
+  tokenData?: number;
+}
+
+export interface Input {
+  txId: string;
+  value: number;
+  tokenData: number;
+  script: string;
+  decoded: DecodedScript;
+  index: number;
+  token?: string;
+}
+
+export interface Output {
+  value: number;
+  tokenData: number;
+  script: string;
+  decoded: DecodedScript;
+  token?: string;
+  spentBy?: string;
+}
+
+export interface Token {
+  uid: string;
+  name: string;
+  symbol: string;
+}
+
+export interface FullTx {
+  txId: string;
+  nonce: string;
+  timestamp: number;
+  version: number;
+  weight: number;
+  parents: string[];
+  inputs: Input[];
+  outputs: Output[];
+  tokens?: Token[];
+  raw?: string;
+}
+
+export interface FullBlock {
+  txId: string;
+  version: number;
+  weight: number;
+  timestamp: number;
+  isVoided: boolean;
+  inputs: Input[];
+  outputs: Output[];
+  parents: string[];
+  tokens?: Token[];
+  height: number;
+}
+
+export interface ApiResponse {
+  success: boolean;
+  message?: string;
+}
+
+export interface DownloadBlockApiResponse extends ApiResponse {
+  block: FullBlock;
+}
+
+export interface SyncSchema {
+  states: {
+    idle: {};
+    syncing: {};
+    failure: {};
+    reorg: {};
+  }
+}
+
+export interface SyncContext {
+  hasMoreBlocks: boolean;
+  error?: {};
+}
+
+export type SyncEvent =
```

Comment:
It is not.. Yet

We are currently using `any` for the events because I was having an issue with typescript types on them

I left a TODO to debug it further because it was taking too much time

Since the event is only locally generated, I don't see it as a big risk for now

Commented it out in 3d3cbed54a2f6b41e486a78355fa2220ea108459